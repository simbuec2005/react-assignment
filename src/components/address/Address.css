.address-container {
    display: flex;
    flex-direction: column;
    flex-basis: 30%;
    padding: 0 20px;
}

.address {
    background-color: #e6e6e6;
    margin-bottom: 5px;
    display: flex;
    padding: 10px;
}

.title {
    margin-bottom: 1rem;
}