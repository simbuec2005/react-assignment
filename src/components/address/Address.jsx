import React from 'react';

import './Address.css';

class Address extends React.Component {
    componentDidMount() {
        const { customer: { id } } = this.props;
        this.props.getCustomerAddress(id);
    }


    render() {

        const { address, customer: { name } } = this.props;
        return (
            <div className="address-container">
                <h3 className="title">Adderesses of {name}</h3>
                {
                    address.loading ? (<div>Loading...</div>) : address.data.length > 0 ? address.data.map((address, index) => {
                        return (
                            <div key={index} className="address">
                                <address>
                                    {address.streetAddress},
                                    {address.city},
                                    {address.state},
                                    {address.postalCode}
                                </address>
                            </div>
                        )
                    }) : <div>No address assigned</div>
                }
            </div>
        )
    }

}

Address.defaultProps = {
    address: {
        data: [],
        loading: false
    }
}

export default Address;