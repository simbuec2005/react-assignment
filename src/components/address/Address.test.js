import React from 'react';
import { create } from 'react-test-renderer';
import { Provider } from 'react-redux'
import { createStore } from 'redux'
import Address from './Address';

const store = createStore(() => { return {}; });

describe('<Address />', () => {
  test('Should match snapshot', () => {
    const props = {
      customer: {
        "id": "123",
        "age": "25",
        "name": "Neil Amstrong"
      },
      getCustomerAddress: ()=> {}
    }
    const component = create(
      <Address {...props} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
})
