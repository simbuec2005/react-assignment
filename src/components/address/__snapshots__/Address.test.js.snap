// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Address /> Should match snapshot 1`] = `
<div
  className="address-container"
>
  <h3
    className="title"
  >
    Adderesses of 
    Neil Amstrong
  </h3>
  <div>
    No address assigned
  </div>
</div>
`;
